### DISCUSSION TITLE
Community call: finding a platform for our video calls

### DISCUSSION THREAD

**Discussion by romanlutz** [fairlearn/fairlearn#777] [General]
So far we've been using Microsoft Teams since the project started out at Microsoft, but since it's a community-driven project now and several people have had issues we wanted to discuss our options. Below are the options I've considered, but I may well have missed some. Feel free to bring in your own perspective, since your experience with these platforms is exactly what we're interested in here.

Whenever something is really bad I've put a :x: to indicate that it basically can't work for us.

| platform | pricing | moderation options | other notes |
| --- | --- | --- | --- |
| Teams (from MSFT tenant) = current solution | no cost | admin can kick anyone out, other members can kick each other out, lobby prevents unwanted guests from joining (but also prevents non-MSFT people from joining) | People have had issues logging in, using Teams on Linux, using all features. |
| Teams (from a non-MSFT tenant) | no cost | only the "owner" of the meeting can let people into the room :x: There don't seem to be configuration options, or at least Teams claimed that I don't have permissions to see them (on my own meeting). | Not really an improvement over Teams from the MSFT tenant since it would also be "owned" by a single person. |
| Zoom | free only for 40 minutes :x: | | | Almost everyone has used Zoom; [security issues](https://www.tomsguide.com/news/zoom-security-privacy-woes) |
| Google Meet | | There don't seem to be any configuration options. :x: | |
| Discord | free (at least 50, but Discord says 99) | Has a very sophisticated moderation system with custom defined roles. One can kick people off the server. This system would allow us to post a Discord widget on our website to allow people to join without requiring us to explicitly add them. | Voice channels could be used for calls or generally meeting other community members. Discord is also popular for sprints, so that would give us a platform to use for sprints as well. Discord has GitHub integration, although I have not set that up yet. |
| Jitsi | Free and no account needed! For up to 100 people | moderation options only available in the paid tier | |

The ideal platform would have the following properties:
- free for our community for at least 60 minutes per call
- moderation capabilities so that we can prevent disruption from potential Zoombombing (e.g. "lobby" and moderator can remove participants); moderation privileges should be shareable and not just held by a single person (since they may not be available in every single session)
- participants can subscribe and unsubscribe from call invites themselves without organizer's approval; ideally we should be able to post the invite on our website and Twitter (only possible in combination with sufficient moderation capabilities)

From my perspective, none of Teams, Zoom, or Google Meet really suit our needs. However, I have never used Discord or Jitsi for calls and can't say whether the experience is good. Any thoughts?
@fairlearn/fairlearn-maintainers @LisaIbanez @michaelamoako @rishabhsamb @LeJit @hannawallach @shimst3r @wcheung-code @brkifle @mesameki @koaning @Dref360

**Comment by adrinjalali** [comment]
- Google meet is actually doesn't let anybody in unless you let them in. I haven't had any issues there. I think people need a google account to join, otherwise it's a pretty descent thing, and I'd say quite okay for community calls.
- Discord isn't really good for only doing calls, it's kinda different. I'm happy to use it for hackathons and stuff, but we really don't need that level of control for a community call.
- We could try jitsi (https://meet.jit.si/)


I'd say we can think about trolls when we have them, and the default settings in the above platforms are pretty okay. We can start with some google meet sessions and see if we like it.

**Reply by romanlutz** [reply]
The problem with "doesn't let anybody in unless you let them in" is that that probably only applies to the meeting "owner", right? So if I'm the "owner" and I can't make the call for one week then we need to create a new meeting.

I don't think the Google account is a big obstacle, but if people think it is I want to take it seriously.

@mmadaio also previously suggested jitsi! Any positive/negative experiences?

**Reply by mmadaio** [reply]
I haven't actually used it, but I know some activist friends have, for organizing, when they didn't want to use a platform owned by a large(r?) tech company

**Comment by shimst3r** [comment]
Some thoughts on Discord: I’ve been playing pen & paper RPG via Discord for a few weeks now and the quality is good enough to get properly immersed. I can’t speak for bigger groups though, as I haven’t experienced one on the platform so far.

Once the community grows, Discord could be a good alternative to eg Gitter, as Gitter's linear message history might get overwhelming quickly for people. Does Gitter have the option to split into multiple chat rooms?

Last thing: I've seen several Discord servers that have implemented a "Code of Conduct Gateway", i.e. when you join a server, you only see a dedicated CoC message that you have to read and react to. Once the server registers your positive reaction, you’ll be granted access to the remaining channels.

**Reply by romanlutz** [reply]
Yes, Gitter allows splitting into multiple rooms, but we haven't had enough activity to warrant that. For a sprint that would be quite useful, though.

**Reply by shimst3r** [reply]
That’s interesting. 🤓 Maybe it would be worthwhile to follow this route then, given you don’t have to install anything, compared to Discord.

**Reply by romanlutz** [reply]
We can definitely do that, but Gitter doesn't support calls. That was the most pressing point. What channels would you like to see?

**Reply by mmadaio** [reply]
I've also used Discord for gaming (TTRPGs too!) and I've found it to be a really good platform for both video and persistent chat (with multiple channels, integrations, etc), and I know they've put a lot of time and money into their moderation capabilities for their gaming audiences. 

One minor quibble is that I do think the way Discord handles cross-talk is not quite as good as Zoom or Teams, but that should be less of an issue in our discussions, with some facilitation, than while playing a game for instance

**Comment by IMPranshu** [comment]
I will recommend (https://www.webex.com/video-conferencing) as I have seen the companies, as well as institutions, are usually preferring this for VC  with the community. I don't know well about the constraints of the platform but I think they are pretty soft on that side. Hope it helps,

**Reply by romanlutz** [reply]
Thanks for the suggestion! I'm a bit reluctant to make people install yet another piece of software. I personally don't use webex for work or otherwise, but if this is common for most others I'm happy to get on it.

**Comment by romanlutz** [comment]
Having used Discord for quite a while now I think we should consider how the experience has been for people so far. Please comment in this thread if you have thoughts! @LeJit @bram49 @adrinjalali @hildeweerts @bramreinders97 @SeanMcCarren @riedgar-ms @MiroDudik @mmadaio @shimst3r @alliesaizan  @lgfunderburk (sorry if I'm omitting people it's likely because I don't know everyone's GitHub handle)

For me personally it took a little while to get used to it with all the knobs and the UI. Now that I am familiar with it I really appreciate having the Discord channels and meeting in one place. Would be great if it was all on GitHub but I guess that's wishful thinking 😄

**Reply by adrinjalali** [reply]
I'm also quite okay now with Discord. Took me a while to get used to it, but I haven't had any issues with the calls really, and it's gotten a lot better in the past year.

**Reply by LeJit** [reply]
I'm fine with Discord too. I've only had a couple of issues with video calls. My only "grip" with Discord is that the chat and the video are separate, but other than that, I'm happy with Discord.

**Reply by alliesaizan** [reply]
I like Discord, and I was already using it before I joined this server. My only real drawback is that there are no reactions like in Teams video chat. I agree that I wish the video came with a chat, but I think the way it's currently set up works fine.

**Reply by hildeweerts** [reply]
Agree with all of the above! I like it much better than our previous Gitter/Teams setup.
